pymc3/variational/inference.py

```diff
@@ -0,0 +1,523 @@
+        See (Sticking the Landing; Geoffrey Roeder,
+        Yuhuai Wu, David Duvenaud, 2016) for details
+
+    References
+    ----------
+    - Kucukelbir, A., Tran, D., Ranganath, R., Gelman, A.,
+        and Blei, D. M. (2016). Automatic Differentiation Variational
+        Inference. arXiv preprint arXiv:1603.00788.
+
+    - Geoffrey Roeder, Yuhuai Wu, David Duvenaud, 2016
+        Sticking the Landing: A Simple Reduced-Variance Gradient for ADVI
+        approximateinference.org/accepted/RoederEtAl2016.pdf
+
+    - Kingma, D. P., & Welling, M. (2014).
+      Auto-Encoding Variational Bayes. stat, 1050, 1.
+    """
+    def __init__(self, local_rv=None, model=None, cost_part_grad_scale=1, gpu_compat=False):
+        super(FullRankADVI, self).__init__(
+            KL, FullRank, None,
+            local_rv=local_rv, model=model, cost_part_grad_scale=cost_part_grad_scale, gpu_compat=gpu_compat)
+
+    @classmethod
+    def from_mean_field(cls, mean_field, gpu_compat=False):
+        """
+        Construct FullRankADVI from MeanField approximation
+
+        Parameters
+        ----------
+        mean_field : MeanField
+            approximation to start with
+
+        Flags
+        -----
+        gpu_compat : bool
+            use GPU compatible version or not
+
+        Returns
+        -------
+        FullRankADVI
+        """
+        full_rank = FullRank.from_mean_field(mean_field, gpu_compat)
+        inference = object.__new__(cls)
+        objective = KL(full_rank)(None)
+        inference.objective = objective
+        inference.hist = np.asarray(())
+        return inference
+
+    @classmethod
+    def from_advi(cls, advi, gpu_compat=False):
+        """
+        Construct FullRankADVI from ADVI
+
+        Parameters
+        ----------
+        advi : ADVI
+
+        Flags
+        -----
+        gpu_compat : bool
+            use GPU compatible version or not
+
+        Returns
+        -------
+        FullRankADVI
+        """
+        inference = cls.from_mean_field(advi.approx, gpu_compat)
+        inference.hist = advi.hist
+        return inference
+
+
+def fit(n=10000, local_rv=None, method='advi', model=None, **kwargs):
+    """
+    Handy shortcut for using inference methods in functional way
+
+    Parameters
+    ----------
+    n : int
+        number of iterations
+    local_rv : dict
+        mapping {model_variable -> local_variable}
+        Local Vars are used for Autoencoding Variational Bayes
+        See (AEVB; Kingma and Welling, 2014) for details
+    method : str or Inference
+        string name is case insensitive in {'advi', 'fullrank', 'advi->fullrank'}
+    model : None or Model
+    frac : float
+        if method is 'advi->fullrank' represents advi part
+    kwargs : kwargs for Inference.fit
+
+    Returns
+    -------
+    Approximation
+    """
+    if model is None:
+        model = pm.modelcontext(model)
+    _select = dict(
+        advi=ADVI,
+        fullrank=FullRankADVI,
```

Comment:
Should stick to a single name that we like. I can see the case against ADVI from the perspective of this PR, but we have used it before and it's not wrong, so I suggest we stick to that.